core/src/main/scala/akka/kafka/internal/MessageBuilder.scala

```diff
@@ -180,7 +180,7 @@ private[kafka] final class CommittableOffsetBatchImpl(
     }
 
     val newOffsets =
-      offsetsAndMetadata.updated(key, new OffsetAndMetadata(committableOffset.partitionOffset.offset, metadata))
+      offsetsAndMetadata.updated(key, new OffsetAndMetadata(committableOffset.partitionOffset.offset + 1L, metadata))
```

Comment:
I do not see immediately, why the change in offset increment and decrement above.

Reply:
The `OffsetAndMetadata` is now passed through as-is. Before it was constructed, decomposed and constructed again with the `+ 1`.